src/BlockTypes/Checkout.php

```diff
@@ -100,6 +98,21 @@ protected function enqueue_data( array $attributes = [] ) {
 			$this->hydrate_from_api( $data_registry );
 			$this->hydrate_customer_payment_methods( $data_registry );
 		}
+
+		Package::container()->get( PaymentMethodAssets::class )->register_payment_method_data();
```

Comment:
Should we still keep an action here for extensions to have a place to enqueue additional data for the block? Also, could `PaymentMethodAssets` just hook _into_ the action?

Reply:
Actually we'll **need to keep** the old action name for checkout and cart blocks (`woocommerce_blocks_enqueue_checkout_block_scripts_after`) because @haszari merged a [pull in jetpack](https://github.com/Automattic/jetpack/pull/15223) that utilizes the hook for tracks.  We can keep the new hook name for the `enqueue_data` methods, but we'll need to keep the old hook name in `enqueue_scripts`.

Reply:
We shouldn't need to because extensions _could_ interact with the registry themselves. I think we should keep this contained within the API we're providing tbh